README.md

```diff
@@ -1,39 +1,601 @@
+
+    ```ruby
+    require 'daru/io' #! Requires all Importers & Exporters
+    require 'daru/io/importers' #! Requires all Importers and no Exporters
+    require 'daru/io/importers/json' #! Requires only JSON Importer
+    ```
+
+**Note: Each IO module has it's own set of dependencies. Have a look at the [Importers](#importers) and [Exporters](#exporters) section for dependency-specific information.**
+
+# Importers
+
+The **Daru::IO** Importers are intended to return a **Daru::DataFrame** from the given arguments. Generally,
+all Importers can be called in two ways - from **Daru::IO** or **Daru::DataFrame**.
 
 ```ruby
-gem 'daru-io'
+#! Partially requires Format Importer
+require 'daru/io/importers/format'
+
+#! Usage from Daru::IO
+Daru::IO::Importers::Format.new(args).call
+
+#! Usage from Daru::DataFrame
+Daru::DataFrame.from_format(args)
 ```
 
-And then execute:
+**Note: Please have a look at the respective Importer Doc links below, for having a look at arguments and examples.**
+
+### ActiveRecord Importer
+
+[(Go to Table of Contents)](#table-of-contents)
+
+Imports a **Daru::DataFrame** from an **ActiveRecord** connection.
+
+- **Docs**: [rubydoc.info](http://www.rubydoc.info/github/athityakumar/daru-io/master/Daru/IO/Importers/ActiveRecord)
+- **Gem Dependencies**: `activerecord` gem
+- **Other Dependencies**: Database(s) such as SQL / Postgresql / etc.
+- **Usage**:
+    ```ruby
+    #! Partially require just ActiveRecord Importer
+    require 'daru/io/importers/active_record'
+
+    #! Usage from Daru::IO
+    Daru::IO::Importers::ActiveRecord.new(activerecord_relation, :field_1, :field_2).call
+
+    #! Usage from Daru::DataFrame
+    Daru::DataFrame.from_activerecord(activerecord_relation, :field_1, :field_2)
+    ```
+
+### Avro Importer
+
+[(Go to Table of Contents)](#table-of-contents)
+
+Imports a **Daru::DataFrame** from an **.avro** file.
+
+- **Docs**: [rubydoc.info](http://www.rubydoc.info/github/athityakumar/daru-io/master/Daru/IO/Importers/Avro)
+- **Gem Dependencies**: `avro` and `snappy` gems
+- **Usage**:
+    ```ruby
+    #! Partially require just Avro Importer
+    require 'daru/io/importers/avro'
+
+    #! Usage from Daru::IO
+    Daru::IO::Importers::Avro.new('path/to/avro/file').call
+
+    #! Usage from Daru::DataFrame
+    Daru::DataFrame.from_avro('path/to/avro/file')
+    ```
+
+### CSV Importer
+
+[(Go to Table of Contents)](#table-of-contents)
+
+Imports a **Daru::DataFrame** from a **.csv** or **.csv.gz** file.
+
+- **Docs**: [rubydoc.info](http://www.rubydoc.info/github/athityakumar/daru-io/master/Daru/IO/Importers/CSV)
+- **Usage**:
+    ```ruby
+    #! Partially require just CSV Importer
+    require 'daru/io/importers/csv'
+
+    #! Usage from Daru::IO
+    Daru::IO::Importers::CSV.new('path/to/csv.gz/file', skiprows: 10, compression: :gzip).call
+
+    #! Usage from Daru::DataFrame
+    Daru::DataFrame.from_csv('path/to/csv.gz/file', skiprows: 10, compression: :gzip)
```

Comment:
It looks a bit confuzing (like file `csv.gz`, and some sub-part of it named `file`), let's use `path/to/file.csv.gz`